model-optimizer/extensions/front/tf/CTCLossReplacement.py

```diff
@@ -78,53 +59,26 @@ def pattern(self):
-                        ' must be specified in a mask format.'.format(ctc_greedy_decoder_tf_name))
-        log.error('The format of input sequence length has been changed to a mask format', extra={'is_warning': True})
-        seq_len_tf_type = seq_len_tf.soft_get('data_type', None)
-        seq_len_tf_name = seq_len_tf.soft_get('name', seq_len_tf.id)
-        seq_mask_placeholder = Parameter(graph, {'name': seq_len_tf_name, 'shape': seq_len_tf_shape,
-                                                 'data_type': seq_len_tf_type}).create_node()
-        reduce_to_seq_len_node = create_op_with_const_inputs(graph, ReduceSum, {1: np.array(1, dtype=np.int32)},
-                                                             {'name': seq_len_tf_name + '/ReduceToSeqLen',
-                                                              'keep_dims': False})
-        reduce_to_seq_len_node.in_port(0).connect(seq_mask_placeholder.out_port(0))
-        seq_len_tf.out_port(0).get_connection().set_source(reduce_to_seq_len_node.out_port(0))
-
-        cast_fp_type = data_type_str_to_np(graph.graph['cmd_params'].data_type)
-        casted_seq_mask_node = Cast(graph, {'name': seq_len_tf_name + '/CastToFP32', 'dst_type': cast_fp_type}).create_node()
-        casted_seq_mask_node.in_port(0).connect(seq_mask_placeholder.out_port(0))
-        permuted_casted_seq_mask = create_op_with_const_inputs(graph, Transpose, {1: int64_array([1, 0])},
-                                                               {'name': seq_len_tf_name + '/Permute'})
-        permuted_casted_seq_mask.in_port(0).connect(casted_seq_mask_node.out_port(0))
-        rename_nodes([(seq_len_tf, seq_len_tf_name + '/AbandonedName'), (seq_mask_placeholder, seq_len_tf_name)])
-
-        # create CTCGreedyDecoder node and set mask node
-        ctc_merge_repeated_i = ctc_greedy_decoder_tf.soft_get('ctc_merge_repeated', ctc_greedy_decoder_tf.id)
-        ctc_greedy_decoder = CTCGreedyDecoderOp(graph, {'name': output_sparse_to_dense_name,
-                                                        'ctc_merge_repeated': ctc_merge_repeated_i}).create_node()
-        ctc_greedy_decoder.in_port(1).connect(permuted_casted_seq_mask.out_port(0))
-        rename_nodes([(sparse_to_dense_tf, output_sparse_to_dense_name + '/AbandonedName'),
-                      (ctc_greedy_decoder, output_sparse_to_dense_name)])
-
-        # create CTCLoss node and set attributes
+        assert ctc_greedy_decoder_tf.has_valid('merge_repeated'), \
+            'The CTCGreedyDecoderSeqLen node "{}" misses "merge_repeated" attribute'.format(ctc_greedy_decoder_tf_name)
+        merge_repeated_tf = ctc_greedy_decoder_tf.merge_repeated
+        ctc_greedy_decoder = CTCGreedyDecoderSeqLenOp(graph, {'name': ctc_greedy_decoder_tf_name,
+                                                              'merge_repeated': merge_repeated_tf}).create_node()
```

Comment:
you don't seem to preserve node name, i.e. move name `CTCGreedyDecoderSeqLenOp` as `SparseToDense`.
Please also check preserving names in CTCGreedyDecoderReplacement. This transformation can break some test or customer's functionality due to this.